packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -316,150 +199,66 @@ class AnalyzeCommand extends FlutterCommand {
-    File optionsFile = new File(path.join(ArtifactStore.flutterRoot, 'packages', 'flutter_tools', 'flutter_analysis_options'));
-    File packagesFile = new File(path.join(host.path, '.packages'))..writeAsStringSync(packagesBody.toString());
-
-    List<String> cmd = <String>[
-      sdkBinaryName('dartanalyzer', sdkLocation: argResults['dart-sdk']),
-      // do not set '--warnings', since that will include the entire Dart SDK
-      '--ignore-unrecognized-flags',
-      '--enable_type_checks',
-      '--package-warnings',
-      '--fatal-warnings',
-      '--fatal-hints',
-      // defines lints
-      '--options', optionsFile.path,
-      '--packages', packagesFile.path,
-      mainFile.path
-    ];
-
-    Status status;
+    String packagesFilePath = path.join(host.path, '.packages');
+    new File(packagesFilePath).writeAsStringSync(packagesBody.toString());
+
     if (argResults['preamble']) {
       if (dartFiles.length == 1) {
-        status = logger.startProgress('Analyzing ${path.relative(dartFiles.first)}...');
+        logger.printStatus('Analyzing ${path.relative(dartFiles.first.path)}...');
       } else {
-        status = logger.startProgress('Analyzing ${dartFiles.length} entry points...');
+        logger.printStatus('Analyzing ${dartFiles.length} files...');
       }
-      for (String file in dartFiles)
-        printTrace(file);
     }
+    DriverOptions options = new DriverOptions();
+    options.dartSdkPath = argResults['dart-sdk'];
+    options.packageConfigPath = packagesFilePath;
+    options.analysisOptionsFile = path.join(ArtifactStore.flutterRoot, 'packages', 'flutter_tools', 'flutter_analysis_options');
+    AnalysisDriver analyzer = new AnalysisDriver(options);
 
-    printTrace(cmd.join(' '));
-    Process process = await Process.start(
-      cmd[0],
-      cmd.sublist(1),
-      workingDirectory: host.path
-    );
-    int errorCount = 0;
-    StringBuffer output = new StringBuffer();
-    process.stdout.transform(UTF8.decoder).listen((String data) {
-      output.write(data);
-    });
-    process.stderr.transform(UTF8.decoder).listen((String data) {
-      // dartanalyzer doesn't seem to ever output anything on stderr
-      errorCount += 1;
-      printError(data);
-    });
-
-    int exitCode = await process.exitCode;
-    status?.stop(showElapsedTime: true);
-
-    List<Pattern> patternsToSkip = <Pattern>[
-      'Analyzing [${mainFile.path}]...',
-      new RegExp('^\\[(hint|error)\\] Unused import \\(${mainFile.path},'),
-      new RegExp(r'^\[.+\] .+ \(.+/\.pub-cache/.+'),
-      new RegExp(r'[0-9]+ (error|warning|hint|lint).+found\.'),
-      new RegExp(r'^$'),
-    ];
-
-    RegExp generalPattern = new RegExp(r'^\[(error|warning|hint|lint)\] (.+) \(([^(),]+), line ([0-9]+), col ([0-9]+)\)$');
-    RegExp conflictingNamesPattern = new RegExp('^The imported libraries \'([^\']+)\' and \'([^\']+)\' cannot have the same name \'([^\']+)\'\$');
```

Comment:
in particular, it would be good to have a test that verifies that we're ok with the author analyzing multiple files that have the same library name.